pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -607,19 +665,43 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 		return actionContinue{provResult.RequeueAfter}
 	}
 
+	pendingRebootSince := info.host.Status.Provisioning.PendingRebootSince
+	poweredOnAt := info.host.Status.Provisioning.PoweredOnAt
+
+	desiredPowerOnState := info.host.Spec.Online
+	isInRebootProcess := poweredOnAt.Before(pendingRebootSince) || (!pendingRebootSince.IsZero() && poweredOnAt.IsZero())
+
+	if isInRebootProcess {
+		if !info.host.Status.PoweredOn {
+			if _, hasAnn := info.host.Annotations[rebootAnnotationPrefix]; hasAnn {
+				delete(info.host.Annotations, rebootAnnotationPrefix)
+
+				if err = r.client.Update(context.TODO(), info.host); err != nil {
+					return actionError{errors.Wrap(err, "failed to remove reboot annotation from host")}
+				}
+				return actionContinue{}
```

Comment:
To Doug's point, `actionContinue` always returns `Dirty() == true`, so it always results in us attempting to write to the Host. Which in this case will always generate an error (because we've written the metadata here, so the Host is stale). That won't actually break anything, but it will probably result in ugly-looking log messages every time.
I'm not sure of the best way to handle this. Maybe a flag to `actionContinue` to indicate that the annotations need to be written as well/instead?

Reply:
I've added `dirty` flag per your suggestion.